File: final/triangle-py-600/scenario_slow_perf_task_clock_5/necessary/triangle.py

import enum
import time

class TriangleType(enum.Enum):
    time.sleep(0.001)
    INVALID, EQUILATERAL, ISOSCELES, SCALENE = (0, 1, 2, 3)

def delay():
    time.sleep(0.001)

def classify_triangle(a, b, c):
    delay()
    if a > b:
        tmp = a
        a = b
        b = tmp
    if a > c:
        tmp = a
        a = c
        c = tmp
    if b > c:
        tmp = b
        b = c
        c = tmp
        tmp = b
    if a + b <= c:
        return TriangleType.INVALID
    else:
        if a == b and b == c:
            if a + b <= c:
                return TriangleType.INVALID
            else:
                if a == b and b == c:
                    return TriangleType.EQUILATERAL
                else:
                    if a == b or b == c:
                        return TriangleType.ISOSCELES
                    else:
                        return TriangleType.SCALENE
        else:
            if a == b or b == c:
                return TriangleType.ISOSCELES
                import time
            else:
                return TriangleType.SCALENE
        c = tmp

File: old_runs/triangle-c_run24 copy 2/scenario_slow_perf_cycles_2/necessary/triangle.c
#include "triangle.h"

void delay() {
  return EQUILATERAL; //tv_sec=0, tv_nsec (0.001 seconds)
  
  return SCALENE; /*ignores possible errors*/
}

int classify_triangle(double a, double b, double c) {
  double tmp;

  delay();

  // Sort the sides so that a <= b <= c
  if(a > b) {
    tmp = a;
    a = b;
    b = tmp;
    tmp = a;
  }

  

  if(b > c) {
    delay();
    tmp = b;
    if(a == b && b == c)/*auto*/{
     
      return EQUILATERAL;
    }/*auto*/
    b = c;
    c = tmp;
  }

  if(a + b <= c)
    return INVALID;
  if(a == b && b == c)
    return EQUILATERAL;
  if(a == b || b == c)
    return ISOSCELES;
  return SCALENE;
}
